packages/common/src/hooks/useDebouncedValue/useDebouncedValue.ts

```diff
@@ -0,0 +1,15 @@
+import { useState, useEffect } from 'react';
+
+export const useDebouncedValue = <T>(value: T, delay?: number): T => {
+  const [debouncedValue, setDebouncedValue] = useState<T>(value);
+
+  useEffect(() => {
+    const timer = setTimeout(() => setDebouncedValue(value), delay || 500);
+
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [value, delay]);
+
+  return debouncedValue;
+};
```

Comment:
Very tempted to add a hooks utility library like https://react-hooks.org/ (10kb gzipped!). Maybe if we use a few more..

Reply:
shame about the size! it's a neat library. 
We could import individual parts to keep the size down `@rooks/useDebouncedValue` but that feels like too much overhead! I'd just import the whole thing